core/src/main/java/org/apache/calcite/plan/AbstractRelOptPlanner.java

```diff
@@ -221,9 +221,13 @@ public long getRelMetadataTimestamp(RelNode rel) {
     return 0;
   }
 
+  @Deprecated
   public void setImportance(RelNode rel, double importance) {
   }
 
+  public void pruneNode(RelNode rel) {
+  }
```

Comment:
Is it possible to add a API to `RelNode`? e.g.
```
void RelNode.setPruned();
boolean RelNode.isPruned();
```

Reply:
I suggest just name it "prune", since it's pretty clear that you are going to prune a RelNode by looking at the parameter. It's also consistent with the original naming (setImportance() not setNodeImportance()).

Reply:
@xndai Revised. Thanks for your good suggestion.